### PR TITLE
Codegen mutation epilogue in _create_runtime_wrapper

### PR DESCRIPTION
**Pull request by bobrenjc93**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #179600

Replace the runtime loop over mutated_inp_runtime_indices with a codegen'd
straight-line function that resolves set_/as_strided_/copy_/detach().copy_
branches at compile time based on each input's mutation metadata.

Updated tests to use requires_grad inputs since inference mutations are
handled inside the graph (keep_input_mutations) and don't use the runtime
epilogue. Removed metadata-only mutation test as transpose_ graph-breaks
through dynamo.

Single data mutation (`x.mul_(2)`):

    def _apply_mutations(orig_inputs, updated_inputs):
        orig_inputs[0].copy_(updated_inputs[0])

Multiple data mutations (`a.mul_(2)`, `c.add_(1)`):

    def _apply_mutations(orig_inputs, updated_inputs):
        orig_inputs[0].copy_(updated_inputs[0])
        orig_inputs[1].copy_(updated_inputs[1])

Leaf mutation under no_grad (`x.detach().mul_(2)`):

    def _apply_mutations(orig_inputs, updated_inputs):
        if orig_inputs[0].requires_grad: orig_inputs[0].detach().copy_(updated_inputs[0])
        else: orig_inputs[0].copy_(updated_inputs[0])

Mutation step in isolation (us/call):

| Case | Before (loop) | After (codegen) | Speedup |
|---|---|---|---|
| 1 mutation / 2 inputs | 11.34 us | 10.96 us | 1.03x |
| 2 mutations / 4 inputs | 22.02 us | 21.95 us | 1.00x |
| 5 mutations / 10 inputs | 54.77 us | 54.12 us | 1.01x |

Performance is dominated by the copy_ calls themselves, so the Python loop
overhead removal is negligible in absolute terms. The primary benefit is
resolving the set_/as_strided_/copy_/detach().copy_ branch at compile time
rather than checking metadata flags per input at runtime.

### CODE REVIEW

**Comment by Lucaskabela** [comment]
@claude please review the test coverage in this PR - focus on edge cases for the types of input mutation that we may have overlooked and odd corner cases. Think hard here

**Review by Lucaskabela — DISMISSED** [review]
Can we add some more test coverage here? I think the change looks good but given the change in runtime_wrappers is ~5 conditionals we should check they get covered properly

**Review by Lucaskabela — DISMISSED** [review]
New test failure seems real (and scary)

**Review by Lucaskabela — APPROVED** [review]
Thanks for updating the logic and additional tests
